### PR TITLE
[2.19.x] DDF-5867 fix table export Visible

### PR DESCRIPTION
**Pull request by andrewzimmer**
#### What does this PR do?
Export Visible always exports the first page of results, this updates table export to use the correct starting index for visible
#### Who is reviewing it? 
@hayleynorton @cassandrabailey293 @zta6 @willwill96 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@bdthomson

#### How should this be tested?
- build / install / run
- ingest data and search for it
- export results via the table
- export Visible results on multiple different pages and verify correct results are present
- hide a result and verify it is not present when exporting visible results
- verify no other regression in exporting

#### What are the relevant tickets?
Fixes: #5867 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java [hunk 1]
```diff
@@ -158,7 +234,48 @@ public Object handle(Request request, Response response) throws Exception {
...
+                    result ->
+                        !cqlTransformRequest
+                            .getHiddenResults()
+                            .contains(result.getMetacard().getId()))
+                .collect(Collectors.toList());
+
+    QueryResponse combinedResponse =
+        new QueryResponseImpl(
+            new QueryRequestImpl(new QueryImpl(ECQL.toFilter(cqlRequests.get(0).getCql()))),
+            results,
```
Comment (cassandrabailey293) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java]: so this `results` is a subset fo the total results, if the results exceeded the size of the request, right? I guess there's no way to only get the number that we want, i.e. `cqlTransformRequest.getCount()` number of results

File: ui/packages/catalog-ui-search/src/main/webapp/extension-points/table-export/table-export.tsx [hunk 2]
```diff
@@ -165,19 +156,30 @@ export const getDownloadBody = (downloadInfo: DownloadInfo) => {
...
+    }
+  })
+
   return {
-    cql,
-    srcs,
+    searches,
     count,
     sorts,
+    hiddenResults,
```
Comment (cassandrabailey293) [inline ui/packages/catalog-ui-search/src/main/webapp/extension-points/table-export/table-export.tsx]: ❓ what's the hidden results for?

Comment (andrewzimmer) [reply ui/packages/catalog-ui-search/src/main/webapp/extension-points/table-export/table-export.tsx]: Results can be hidden on the ui and we don't want these to appear for an export `visible`. Previously `getActiveSearchResults` did this filtering by only retrieving ids of non-hidden results, now we are passing this list of ids to do this filtering on the backend

**Comment by bdeining** [comment]
build now

**Comment by bdeining** [comment]
build now

**Comment by andrewzimmer** [comment]
build now

**Comment by cassandrabailey293** [comment]
hero successful! 🎉 
Verified that exporting visible results will export the correct number and order of results
No regressions found
Changed pages and number of results per page, and verified the correct results were being exported
Hidden results are not exported

**Comment by bdeining** [comment]
build now

**Comment by andrewzimmer** [comment]
build now

**Comment by andrewzimmer** [comment]
build now

**Pull request merged**